https://issues.redhat.com/browse/ACM-20400 Right-sizing doc (#7970)
open-cluster-management/rhacm-docs: observability/obs_right_size_config.adoc

> [config-right-size]

Reviewer: I don't see any reference to the `observability/obs_right_size_config.adoc` file in the current PR(expect entry in `observability/main.adoc`). Not sure how or from where this file’s content will be accessed or rendered. Just wanted to make sure we are not missing anything here.

Author: It shows up for me, can you check again?

![image](https://github.com/user-attachments/assets/53552832-4c18-4615-a9ad-4f850b6b5fcc)

Reviewer: Yes that changes showing up for me as well.  was expecting something like below text but could not found so thought to cross check with you . 
- *Technology Preview:* Use right-sizing recommendations to optimize workloads on a cluster and namespace level. To learn more, see xref:../observability/obs_right_size.adoc#optimize-work-right-size[Optimizing workloads by using right-sizing recommendations (Technology Preview)].

Author: Oh I see, we only need to link to the top level doc, the config doc comes after that if you scroll in the published version.

= Configuring right-sizing recommendations (Technology Preview)

> You can customize right-sizing recommendations by editing the `rs-namespace-config` ConfigMap.

Reviewer 2: ```suggestion
You can customize right-sizing recommendations by editing the `rs-namespace-config` `ConfigMap` resource.
```

[#config-right-size-prereq]
== Prerequisites

- You have enabled right-sizing recommendations.

[#config-right-size-placement]
== Configuring placement

> You can customize the placement configuration by making changes to the `placementConfiguration` propoerty in the `rs-namespace-config` ConfigMap.

Reviewer 2: ```suggestion
You can customize the placement configuration by making changes to the `placementConfiguration` property in the `rs-namespace-config` `ConfigMap` resource.
```

See the following example where the policy only applies to clusters with the `environment=prod` label:

[source,yaml]
----
placementConfiguration: |
 spec:
   . . .
   predicates:
   - requiredClusterSelector:
       labelSelector:
         matchLabels:
           environment: prod
   . . .
----

[#config-right-size-prometheus]
== Configuring `PrometheusRule`

You can customize the `PrometheusRule` resource by making changes to one of the following properties:

`namespaceFilterCriteria`:: Limit recommendations to specific namespaces. You can include or exclude this property.

> `nlabelFilterCriteria`:: Filter namespaces with custom labels.

Reviewer 2: Based on the example yaml, this might be a typo but please confirm!
```suggestion
`labelFilterCriteria`:: Filter namespaces with custom labels.
```

Reviewer 3: this might be correct if they need the label for that namespace `n`

Author: Let me check, I think it needs the n

Author: You were right it's just a typo, updated

`recommendationPercentage`:: Adjust CPU and memory recommendation percentages.

See the following example:

[source,yaml]
----
. . .
  prometheusRuleConfig: |
   namespaceFilterCriteria:
    inclusionCriteria: <1>
    - prod.*

>     exclusionCriteria:

Reviewer 2: ```suggestion
    exclusionCriteria: <1>
```

    - openshift.*
  labelFilterCriteria:
  - labelName: label_kubernetes_io_metadata_name
    inclusionCriteria:
    - prod
    - staging
    exclusionCriteria:
    - kube.*
  recommendationPercentage: 120 <2>
  . . .
----

<1> Use either `inclusionCriteria` or `exclusionCriteria`. You cannot use both at the same time.
<2> Set the percentage of actual usage.

[config-right-size-namespace]
== Configuring namespace binding

> By default, the right-sizing feature uses the `open-cluster-management-global-set` for binding policies to clusters.

Reviewer: Can we include namespace after `open-cluster-management-global-set` for more clarity?

Author: Updated

> To bind generated policies to a different namespace and ClusterSet, use the `namespaceBinding` property. See the following example:

Reviewer 2: Or `ClusterSet` resource/object
```suggestion
To bind generated policies to a different namespace and cluster set, use the `namespaceBinding` property. See the following example:
```

Reviewer 3: Code ClusterSet throughout

[source,yaml]
----
apiVersion: observability.open-cluster-management.io/v1beta2
kind: MultiClusterObservability
metadata:
  name: observability
spec:
  . . .
  capabilities:
    platform:
      analytics:
        namespaceRightSizingRecommendation:
          enabled: true
          namespaceBinding: <your-namespace-binding>
  . . .
----

> *Important:* Ensure that the target namespace is part of a valid ClusterSet before applying the change. The generated policies might not be applied to any clusters if there is no valid binding.

Reviewer 3: ```suggestion
*Important:* Ensure that the target namespace is part of a valid `ClusterSet` before you apply the change. The policies that you generate might not apply to any clusters if there is no valid binding.
```
